### PR TITLE
boards: move remaining uses of USEMODULE from Makefile.include to Makefile.dep

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR moves all remaining uses of `USEMODULE += <module>` from boards Makefile.include to the corresponding Makefile.dep.

### Testing procedure

- A green Murdock
- for each modified board, check the list of modules included in a build and compare to master

### Issues/PRs references

#9913

### CODE REVIEW

File: boards/arduino-mega2560/Makefile.include [hunk 1]
```diff
@@ -1,5 +1,3 @@
-USEMODULE += boards_common_arduino-atmega
```
Comment (smlng) [inline boards/arduino-mega2560/Makefile.include]: did this one got lost? or did I overlooked it?

**Comment by gschorcht** [comment]
> * for each modified board, check the list of modules included in a build and compare to master

IMHO, we should use again `dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh`. I will try it later today.

**Comment by aabadie** [comment]
Wait a bit @gschorcht, Murdock is reporting a lot of failures. I'd like to fix them first.

**Comment by gschorcht** [comment]
@aabadie I have the script already script running, but for current master first. This will be needed anyway.

**Comment by gschorcht** [comment]
> Wait a bit @gschorcht, Murdock is reporting a lot of failures. I'd like to fix them first.

As I have seen only for 4 boards `acd52832`, `ruuvitag`, `wsn430-v1_3b`, `wsn430-v1_4` and for only one test `esp32-olimex-evb`.

File: boards/arduino-mega2560/Makefile.include (hunk 1 above)
Comment (aabadie) [reply boards/arduino-mega2560/Makefile.include]: no idea, did you change this recently ?

**Comment by aabadie** [comment]
@gschorcht, I pushed a fix that I hope should improve the situation.

**Comment by gschorcht** [comment]
> @gschorcht, I pushed a fix that I hope should improve the situation.

I will take a look to the problem with the `esp32-olimex-evb` board.

**Comment by gschorcht** [comment]
@aabadie I figured out a problem for all ESP32 boards in `board.h` headers where `board_common.h` was included inside the `extern "C"` linkage block. Even though PR #13301 fixes the problem, the reason the compilation error occurred with this PR and not in the master is a wrong module:

The `esp32-olimex-32` has an Ethernet interface which should be used default netdev. That's why module `esp_eth` is enabled if the `netdev_default` module is defined. However, depending on this module, some compile options must be set in `Makefile.include`.

And here we encounter the problem of the order in which `Makefile.include` and `Makefile.dep` are read. It must be known in `Makefile.include` whether `esp_eth`, `esp_wifi` or `esp_now` are used in order to set some compile options correctly. However, the dependencies in `Makefile.dep` are read after `Makefile.include` has been read. Therefore, all dependencies for these modules must be kept in `Makefile.include` for the moment until we change this order.

Maybe we could this problem using `FEATURES_OPTIONAL` and `FEATURES_PROVIDED`.

**Comment by aabadie** [comment]
@gschorcht, I'll remove the changes related to the esp8266 board for the moment.

I agree that having features for esp_eth, esp_wifi, esp_now could help with the dependency resolution.

**Comment by aabadie** [comment]
No more build failures on Murdock, OK to squash @gschorcht ?

**Comment by gschorcht** [comment]
Should we first have the results of `dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh`?

**Comment by aabadie** [comment]
> Should we first have the results of dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh?

That would help indeed. Let's wait for the results.

**Comment by gschorcht** [comment]
I ran `dist/tools/buildsystem_sanity_check/save_all_dependencies_resolution_variables.sh` for this PR and the current master. The output generated with
```
BOARD="${board}" make -C "${RIOTBASE}/${application}" dependency-debug
```
is completely identical. However the output from
```
DEPENDENCY_DEBUG=1 make -C "${RIOTBASE}/${application}" info-boards-supported
```
differs slightly. But the only difference is that the list of used module in `USEMODULE` contains an additional entry `boards_common_*` with this PR for the boards that use a common board definition, for example `boards_common_nucleo` for `nucleo-*` boards.

Without having a deep insight why the `info-boards-supported` differs, it seems that the dependencies per application and board are exactly the same. Therefore, I would say that everything is fine.

**Comment by gschorcht** [comment]
@aabadie Please squash.

**Comment by aabadie** [comment]
Done

**Comment by aabadie** [comment]
It's all green @gschorcht !

**Comment by gschorcht** [comment]
@aabadie Thanks for contribution

**Pull request merged**